oauth2/oauth2-resource-server/src/main/java/org/springframework/security/access/expression/AuthenticationResolver.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2002-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.access.expression;
```

Comment:
`AuthenticationResolver` may be useful in other areas of the framework, not just SpEL. So I'm wondering if this belongs in `security-core` and package `org.springframework.security.core`